### PR TITLE
Protect node.ents and node.entsCached with a mutex in fs/layer/node.go

### PR DESCRIPTION
**Pull request by iain-macdonald**
**Issue #, if available:**
None

**Description of changes:**
We run a modified version of the soci-snapshotter in production at BuildBuddy and recently enabled the golang race detector in our dev environment. It flagged racy accesses to `node.ents` and `node.entsCached`, I've attached the [logs](https://github.com/awslabs/soci-snapshotter/files/12591454/datarace.txt) for line numbers (which should be unmodified). This PR protects access to those two fields with a mutex.

**Testing performed:**
In addition to the PR checks, I pulled and ran an indexed image and it seemed to work fine, but didn't do any additional testing beyond that.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Kern--** [comment]
I'd like to take a quick look at benchmarking data before merging this to see if we should be using a `sync.Value` or a more complicated variant of `sync.Once` that allows retries on errors.

**Review by Kern-- — APPROVED** [review]
Sorry it took me a while to get back to this. I'm not seeing any significant difference in our benchmark images.

**Comment by iain-macdonald** [comment]
> Sorry it took me a while to get back to this. I'm not seeing any significant difference in our benchmark images.

Thanks, Kern! Looks like I need another approver to merge. Is there anyone you'd suggest?

**Review by turan18 — APPROVED** [review]

**Pull request merged**
